core/src/main/bin/questdb.sh

```diff
@@ -179,6 +179,14 @@ function start {
     -XX:+UseParallelGC
     "
 
+    if [ "$(uname)" == "Darwin" ]; then
+        N_MAX_OPEN_FILES=$(sysctl -n kern.maxfilesperproc)
```

Comment:
I think we should change this:
```
    N_MAX_OPEN_FILES=$(sysctl -n kern.maxfilesperproc)
    ulimit -n $N_MAX_OPEN_FILES
```
to the following:
```
    # JVM on MacOS has its own max open files limit, set to 10240
    # This limit can be removed by passing the -XX:-MaxFDLimit option
    # However, if this built-in limit is removed, the JVM starts to use the soft limit as if it was the hard limit,
    # so we should set the soft limit to the same value as the hard limit
    ulimit -n $(ulimit -H -n)
```

Reply:
@glasstiger Thanks for comments! I've learned a lot from your experiment. I agree with your opinion.